QubitServer/src/main/java/org/labrad/qubits/Device.scala

```diff
@@ -0,0 +1,96 @@
+package org.labrad.qubits
+
+import com.google.common.collect.Lists
+import com.google.common.collect.Maps
+import java.util.List
+import java.util.Map
+import org.labrad.qubits.channels.Channel
+import scala.collection.JavaConverters._
+
+/**
+ * Model of a device, essentially a collection of named channels of various types.
+ * Each channel connects various physical devices (FastBias, Preamp, GHzDac) to some
+ * controllable parameter on the device (flux bias, microwave, etc.).
+ * 
+ * @author maffoo
+ */
+class Device(name: String) {
+
+  private val channels: List[Channel] = Lists.newArrayList()
+  private val channelsByName: Map[String, Channel] = Maps.newHashMap()
+
+  def getName(): String = {
+    name
+  }
```

Comment:
Truly, the Scala gods are enlightened and benevolent. This just looks so, so weird compared with other languages.